### ISSUE TITLE
DBText::ContextSummary cuts in the middle of line breaks

### ISSUE THREAD

**Issue by mxmclaudel** [silverstripe/silverstripe-framework#9114]
## Affected Version

silverstripe/cms 4.4.0

## Description

When calling DBText::ContextSummary(200). The text returned is plain except for line breaks and `<mark>.` The `<br />` tag gets cut in the middle if if happens to be at the beginning/end of the section returned. It's probably being counted as 2 separate words instead of one.
This results in invalid HTML returned.

**Comment by caffeineinc** [comment]
Hi @mxmclaudel  

Could you please provide an example string which is being cut incorrectly? 

Thanks,

**Comment by ScopeyNZ** [comment]
My guess would be 197 characters of whatever followed by `<br />` will cause it to be cut incorrectly...

**Comment by mxmclaudel** [comment]
Hi @caffeineinc 
Sure, here's the raw HTML from the CMS:

```
<p>This is a lot of text before this but really, this is a test sentence</p>
<p>with about more stuff after the line break</p>
```

In the template I have this:
`$Content.ContextSummary(35, "test")`

In the output page, I get the following HTML:
```
                    ... really, this is a <mark>test</mark> sentence<br...
```

Thanks

**Comment by caffeineinc** [comment]
Hey @mxmclaudel 

I've made a bugfix & PR which should resolve your issue.

**Comment by robbieaverill** [comment]
Merged into 4.5.x-dev
